### PR TITLE
EL-2122 - Infinite Scroll Enhancement

### PR DESCRIPTION
**Pull request by ashley-hunter**
Adding ability to bind to window scroll

- Added logic for handling scroll event on document
- Allowing scrollElement to be an element or ElementRef as document.documentElement is not readily available as an ElementRef.
- Removed onScroll and onDomChange functions as they just called the check function anyway.
- Updated docs with instruction on how to target window scroll

### CODE REVIEW

**Comment by ashley-hunter** [comment]
Here is an example showing the new functionality in action: https://plnkr.co/edit/uXoGCYuHdsf2NU0svkr9?p=preview

**Comment by ashley-hunter** [comment]
Added protractor tests for fullscreen infinite scroll (also removed some test cases from existing infinite scroll tests as they were irrelevant - eg. testing whether or not an accordion expanded/collapsed on click)

File: e2e/tests/components/infinite-scroll/fullscreen/infinite-scroll-fullscreen.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+        // change page size to 39
+        page.clickOnPageSize();
+        page.getPageSize().sendKeys(Key.DELETE);
+        page.getPageSize().sendKeys('3'); // 39
+        expect<any>(page.getNumberOfEmployees()).toEqual(39);
+        page.hoverOverLastEmployee();
+        expect<any>(page.getNumberOfEmployees()).toEqual(78);
+    });
+
+    it('should have min and max page size values', () => {
```
Comment (donna-marijne) [inline e2e/tests/components/infinite-scroll/fullscreen/infinite-scroll-fullscreen.e2e-spec.ts]: I think this test can be removed too.

**Comment by ashley-hunter** [comment]
Tests now removed

**Pull request merged**